Build a regression model.

In [1]:
import numpy as np
import pandas as pd
import csv
from sklearn import linear_model, datasets
import statsmodels.api as sm



In [2]:
df_join = pd.read_csv("bikes_foursquare_join.csv")
df_join.head()

latitude  longitude  empty_slots  free_bikes  total_bike_slots  \
0  54.604948  -5.921534           10           5                15   
1  54.600337  -5.905516            9           6                15   
2  54.595315  -5.949091            8           2                10   
3  54.595840  -5.929881            9           5                14   
4  54.580556  -5.920504            4           6                10   

   count_of_restaurants  
0                    50  
1                    42  
2                    30  
3                    50  
4                    47

In [3]:
X = pd.DataFrame(df_join['count_of_restaurants'])
y = pd.Series(df_join['total_bike_slots'])


In [4]:
# I selected my output to be the total_bike_slots column, because of this I removed both the free_bikes and empty_slots from my input as they are not independent from my output data or column
X = pd.DataFrame(df_join[['latitude', 'longitude', 'total_bike_slots', 'empty_slots', 'free_bikes']])
y = pd.Series(df_join['count_of_restaurants'])

In [5]:
#Including the constant column to my X variable
X=sm.add_constant(X)
lin_reg = sm.OLS(y,X)

In [6]:
lin_model = lin_reg.fit()
print_model = lin_model.summary()
print(print_model)

                             OLS Regression Results                             
Dep. Variable:     count_of_restaurants   R-squared:                       0.106
Model:                              OLS   Adj. R-squared:                  0.040
Method:                   Least Squares   F-statistic:                     1.601
Date:                  Wed, 22 May 2024   Prob (F-statistic):              0.187
Time:                          01:08:40   Log-Likelihood:                -210.92
No. Observations:                    59   AIC:                             431.8
Df Residuals:                        54   BIC:                             442.2
Df Model:                             4                                         
Covariance Type:              nonrobust                                         
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
const              1

Provide model output and an interpretation of the results. 

1. The Adjusted R-squared value is 0.040, meaning my multivariate model explains only 4% of the variations in data
2. For all my data the P-value is greater than 0.05. This means that the relationship between my output and my variable is due to natural variation. 
3. What I can tell from this output is that the longitude has the strongest negative impact on the count of restaurants. Also all my variables had a negavive impact on the count of restaurant output data.


# Stretch

How can you turn the regression model into a classification model?

A good way to turn my model from a regression model to a classification model is to make the total count of restaurant the output or "Y". Then for all count of restaurants values that are 50, rename them as "max", and for count of restaurants values that are not 50, name them as "not max". Then pass them into the model to determine if the X avlues of Dataframe can predict if the y or output would be at max or not max.